File: Activities Python/SortBubbleInsertion.py
def insertionSort(a): #Los elementos del arreglo definida en la funcion
        for i in range(1, len(a)): #Para i, recorrera desde la pos 0 hasta la longitud del arreglo
            temp = a[i] #temp es igual a la pos actual 

            j = i -1 #j es igual a una pos antes de i
            while j >= 0 and temp <a[j] : #Mientras j sea igual o mayor a 0 y temp sea menor
                a[j + 1] = a[j] #Movera a una pos a la derecha
                j = j-1 #Se mueve un paso hacia atras para continuar comparando
            a[j + 1] = temp #Cuando encuentra la pos correcta lo inserta en el temp
def printArr(a): #Lee el arreglo y lo ilustra en la terminal
     for i in range(len(a)):
          print(a[i], end = " ")

a = [70, 12, 32, 34, 65] #Define el arreglo
print("Antes de ordenar los elementos del arreglo:")
printArr(a)
insertionSort(a)
print("\nDespues de arreglos los elemntos del arreglo:")
printArr(a)

File: Activities Java/MethodsOfOrden/SortBubbleInsertion.java
public class SortBubbleInsertion {

    public static void insertionSort(int[] a) {
        for (int i = 1; i < a.length; i++) {
            int temp = a[i]; 
            int j = i - 1;

            while (j >= 0 && temp < a[j]) {
                a[j + 1] = a[j];
                j--;
            }
            a[j + 1] = temp; 
        }
    }

    public static void printArr(int[] a) {
        for (int i = 0; i < a.length; i++) {
            System.out.print(a[i] + " ");                           
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] a = {70, 12, 32, 34, 65};

        System.out.println("Antes de ordenar los elementos del arreglo:");
        printArr(a);

        insertionSort(a);

        System.out.println("Después de ordenar los elementos del arreglo:");
        printArr(a);
    }
}
